### ISSUE TITLE
A question regarding the implementation of `DC_BCE_masked_loss`

### ISSUE THREAD

**Issue by One-punch24** [Fjr9516/DA_nnUNet#2]
Dear Authors,

Thank you so much for this nice idea and nice implementation based on nnunet framework.

I have one specific question about the code for `DC_BCE_masked_loss`  in [losses.py, L96](https://github.com/Fjr9516/DA_nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/customized/losses.py#L96), which is the loss for domain adaptation training.

Well, the parameter `target_domain` of the forward function is a list, instead of `torch.tensor`, for example `[tensor([0, 1, 0, 1], device='cuda:0')]` when I print it. This will cause `domain_target == self.masked_label` to be `False`. From my understanding, it seems we are running `mask[False, :]=0`. 

To the best of my knowledge, Line 96 is to modify the `mask` such that it can mask the computed segmentation loss for target domain. However, I'm not sure whether Line 96 can play this role. 

I would truly appreciate it if you could answer my question. Thank you again for providing this codebase:)!

**Comment by SimoneBendazzoli93** [comment]
Hi @One-punch24 , thank you for pointing this out!

You're absolutely right in identifying that the purpose of Line 96 is to mask out the segmentation loss for samples from the target domain during domain adaptation training.

Regarding your question: in the training loop, target_domain is passed as a torch.Tensor, even if it might appear wrapped in a list (e.g., `[tensor([0, 1, 0, 1], device='cuda:0')]`). In practice, this is handled correctly—when the comparison domain_target == self.masked_label is executed, it produces the expected boolean mask across the batch dimension. So the masking behavior works as intended.

Also, you're right that the line `mask[domain_target == self.masked_label, :] = 0` is a bit unsafe stylistically, especially for 5D tensors. While it does work in this specific case, it's not the most readable or robust way to index. We recommend replacing it with:

`mask[domain_target == self.masked_label, ...] = 0
`
This version is more explicit and safe when working with high-dimensional tensors like [B, C, H, W, D].

Thanks again for your question and for your interest in the codebase!

**Comment by One-punch24** [comment]
Hi, thank you for your answer.

However, actually, I printed the mask in your code and I made a simple statistics with the mask. The mask is all ones. 

Furthermore, I'm quite sure that you passed a list instead of a tensor, through the printing. If you don't believe, you can try to print `type(target_domain)`

**Comment by One-punch24** [comment]
Actually, the phenomenon that led me to discover this issue, was the performance on the target domain before starting the DANN training was already "too perfect", which suggested a leakage of label in target domain.

**Comment by Fjr9516** [comment]
Hi @One-punch24, 
Thank you for pointing this out. A fix has been applied and is available in the latest commit. 
The issue only affects training when you disable deep supervision, other trainers should work as expected. We're currently rerunning the experiments to evaluate the impact and will follow up with any further updates. We really appreciate your attention to detail, and please feel free to open another issue if you come across anything else.
Thanks again for your feedback! :)

**Comment by One-punch24** [comment]
Many thanks!! I would appreciate it if you would like to share the updated results and parameter setting later. 

Thank you again for this nice implementation!
